markdown/how.md

```diff
@@ -3,17 +3,87 @@
+
+The Docker registry supported by Magnum is just the official Docker registry image with the OpenStack Swift storage driver. When invoked by Magnum, it is Magnum that automatically populates the registry's storage configuration with the necessary parameters to match the Swift endpoint in the region it's running in. In our CI, we duplicate this for a locally-running registry instance, backed by the same storage parameters, that we push our images to. Once we do, any thus-registered image becomes available to our Magnum registries.
+
+In our case it's not native OpenStack Swift that backs the registry, but Ceph radosgw with Keystone integration.
+
+We don't just rely on a single registry, but instead, we have two registries one for each region. All images are available in both regions, ensuring redundancy and availability across our infrastructure.
+
+
 
 ## Kubernetes
 
+<!-- Note -->
+
+Magnum lets you create clusters via the OpenStack APIs. To do that, you base your configuration on a cluster template.
+The template defines paramters how the cluster will be constructed.
+
+We deploy our kubernetes cluster with the following configuration.
+In our production environment, we have six node Kubernetes clusters, three control plane nodes, and three worker nodes.
+
+We deploy our cluster with kubernetes version as v.1.21.10 on Fedora CoreOS 34 with Cinder csi driver enabled.
+ Openstack cinder is a block storage service that allows you to create persistent volumes for your kubernetes cluster. By enabling cinder csi driver on your magnum cluster you can create persistent volumes backed by cinder block storage.
+
+Octavia is the OpenStack load balancing service. We create a Kubernetes cluster with Magnum and use an Octavia Load Balancer to access it from the public network, via the the load balancer’s floating IP.
 
 ## Ceph
 
+<!-- Note -->
+
+To achieve automatic replication between the primary and DR site, we use Backup and Restore policy using the Ceph object gateway S3 API. We don't use the multi-site replication as provided by Ceph natively.
+
+In our kubernetes environment, the backup and restore process is implemented using cron jobs.
+
+In Kubernetes, you can achieve similar functionality to the cron job utility in Linux by using CronJobs. CronJobs allow you to schedule and run recurring tasks or jobs within your Kubernetes cluster.
+
+With CronJobs, you can define a schedule using a cron-like syntax, specifying the desired frequency and timing for your tasks. Kubernetes will then automatically create and manage pods to execute those tasks based on the defined schedule.
+In our environment, the cron job is responsible for capturing MySQL and MongoDB dumps from the primary site, uploading them to S3 buckets, and subsequently restoring them from these S3 buckets at the DR site.  
+
+So, in case of any incident, we can do a failover to the DR site for uniterrupted services.
+
```

Comment:
Drop this blank line.

Reply:
Dropping this blank line messes with the layout. I see notes inside the slides.